### PR TITLE
[device/accton] Replace os.system and remove subprocess with shell=True

### PR DESCRIPTION
**Pull request by maipbui**
Signed-off-by: maipbui <maibui@microsoft.com>

#### Why I did it
`subprocess.Popen()` and `subprocess.run()` is used with `shell=True`, which is very dangerous for shell injection.
`os` - not secure against maliciously constructed input and dangerous if used to evaluate dynamic content
#### How I did it
Replace `os` by `subprocess`
Remove unused functions
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [X] 201911
- [X] 202006
- [X] 202012
- [X] 202106
- [X] 202111
- [X] 202205

### CODE REVIEW

File: device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/sfp.py [hunk 1]
```diff
@@ -60,7 +59,7 @@ def get_eeprom_path(self):
         return self.port_to_eeprom_mapping[self.port_num]
 
     def __is_host(self):
-        return os.system(self.HOST_CHK_CMD) == 0
+        return subprocess.run(self.HOST_CHK_CMD, stdout=None, stderr=None).returncode == 0
```
Comment (qiluo-msft) [inline device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/sfp.py:62]: >, stdout=None, stderr=None [](http://example.com/codeflow?start=47&length=26)

You are using default values for both parameters. So you can remove them.

**Comment by qiluo-msft** [comment]
>         return self.__run_command(install_command)

It is not defined. @simonJi2018 Could you check whether it is a syntax error in original code?

---
Refers to: device/accton/x86_64-accton_as7116_54x-r0/sonic_platform/component.py:78 in 69b7f29. [](commit_id = 69b7f29915bf44592432643331d9288c6695406e, deletion_comment = False)

**Review by qiluo-msft — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by maipbui** [comment]
@ec-michael-shih @jeffhtgt @seanwu-ec @jostar-yang Could you help review and verify?

**Pull request merged**